### ISSUE TITLE
Non-rendered input files for HTCondor workflows

### ISSUE THREAD

**Issue by tvoigtlaender** [riga/law#119]
As far as I can see, there is currently no way to add files that should not be 
rendered to the `transfer_input_files` parameter of an HTCondor job file. 

This is especially detrimental for `.tar.gz` files as the python3 `open().read()` is unable to read them. 
https://github.com/riga/law/blob/master/law/job/base.py#L735
(This did not result in a fatal error in python2, but does in python3.)

There are some workarounds to this (like the `errors="ignore"` argument of `open()`), 
but an option to append un-rendered files to the list of rendered ones in the job file would be helpful.

**Comment by tvoigtlaender** [comment]
Unfortunately it seems like the workaround I mentioned breaks the `.tar.gz` file.

**Comment by riga** [comment]
Hi @tvoigtlaender ,

Good point! I just added a few changes in a feature branch https://github.com/riga/law/tree/feature/improve_job_input_control .

- There is now a try-except clause around `f.read()` that checks for `UnicodeDecodeError` errors and silently returns by default.
- To have more control about what's done with input files, there is now a shallow class `law.JobInputFile` that accepts the input path, as well as the flags `copy`, `render`, `postfix` which are all true by default.

Could you give it a try with

```python
config.input_files["your_tgz_file"] = law.JobInputFile("your.tar.gz", render=False)
```

?

**Comment by riga** [comment]
Could you maybe send me a bundle with all files belonging to a single job? I think this would help me debugging this :) (CERN Mattermost for instance). Thank you!

**Comment by riga** [comment]
Fixed in #120 🎉
